src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerAmendCase.java

```diff
@@ -0,0 +1,59 @@
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.divorce.caseworker.event.page.AmendCase;
+import uk.gov.hmcts.divorce.common.ccd.CcdPageConfiguration;
+import uk.gov.hmcts.divorce.common.ccd.PageBuilder;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AosDrafted;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AosOverdue;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingAos;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingClarification;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingConditionalOrder;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingLegalAdvisorReferral;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingPronouncement;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.ConditionalOrderDrafted;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.ConditionalOrderPronounced;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.ConditionalOrderRefused;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.DefendedDivorce;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.Holding;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASEWORKER_COURTADMIN_CTSC;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASEWORKER_LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASEWORKER_SUPERUSER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.READ;
+
+@Component
+public class CaseworkerAmendCase implements CCDConfig<CaseData, State, UserRole> {
+    public static final String CASEWORKER_AMEND_CASE = "caseworker-amend-case";
+    private final CcdPageConfiguration amendCase = new AmendCase();
+
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        var pageBuilder = addEventConfig(configBuilder);
+        amendCase.addTo(pageBuilder);
+    }
+
+    private PageBuilder addEventConfig(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        return new PageBuilder(configBuilder
+            .event(CASEWORKER_AMEND_CASE)
+            .forStates(AwaitingAos, AosDrafted, AosOverdue,
+                DefendedDivorce, Holding, AwaitingConditionalOrder,
+                ConditionalOrderDrafted, AwaitingLegalAdvisorReferral, AwaitingClarification,
+                ConditionalOrderRefused, AwaitingPronouncement, ConditionalOrderPronounced)
+            .name("Update Case")
+            .description("Update Case")
+            .explicitGrants()
+            .grant(CREATE_READ_UPDATE,
+                CASEWORKER_COURTADMIN_CTSC)
```

Comment:
do we also need `CASEWORKER_COURTADMIN_RDU` ?

Reply:
Story states only about - caseworker-courtadmin

Reply:
court admin CTSC and courtadmin RDU ?